src/session/components/message_entry.rs

```diff
@@ -216,6 +229,33 @@ impl MessageEntry {
         self.imp().text_view.buffer().set_text(&text);
     }
 
+    pub(crate) async fn parsed_formatted_text(&self) -> Option<FormattedText> {
```

Comment:
Maybe we should call this function just `as_markdown` to give it a more precise name.